src/logic/wallets/getWeb3.js

```diff
@@ -36,12 +37,23 @@ export const openTxInEtherScan = (tx: string, network: string) => `https://${net
 export const getEtherScanLink = (address: string, network: string) => `https://${network}.etherscan.io/address/${address}`
 
 let web3
-export const getWeb3 = () => web3 || new Web3(window.web3.currentProvider)
-
-const isMetamask: Function = (web3Provider): boolean => {
-  const isMetamaskConstructor = web3Provider.currentProvider.constructor.name === 'MetamaskInpageProvider'
+export const getWeb3 = () => web3
```

Comment:
why was this changed? I have no idea why a condition was there but maybe for testing purposes? Does it work when testing?
you can run these tests:
```
yarn test src/test/safe.dom.load.test.js 
yarn test src/test/safe.dom.create.test.js 
```
(migrate safe-contracts first)